### PR TITLE
Better handling of config loading when using CLI tools

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

When we're running CLI tools, we get this annoying message about failing to initialize SlackAppender. This PR addresses that issue by fixing handling of config loading. Essentially, the idea is to handle nulls in our static blocks. See comments for more details.

_What about production config loading in study-server and housekeeping?_
* These use the config extensively, so we will have gotten NullPointerException very early in the process if config is not provided.

_What about the study-builder CLI tool?_
* I tried it out locally and study-builder CLI works as normal.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

- [x] :relaxed: All good, business as usual!

## How do we demo these changes?

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/ConfigManager.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ private ConfigManager(Config cfg) {
      * @return the config for the app
      */
     public static synchronized ConfigManager getInstance() {
-        if (configManager == null) {
+        if (configManager == null && TYPESAFE_CONFIG_FILE != null) {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/ConfigManager.java:57]: When we're running CLI tools (like `AesUtil`), the first thing that runs is the SlackAppender constructor, which calls this `getInstance()` method to grab the config object. Somehow, the `static` block in this file doesn't get run, which means `TYPESAFE_CONFIG_FILE` variable is null. So we add null handling here. Idea here is: if there's no config file, we won't initialize the config manager, so callers need to handle null config managers.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/log/SlackAppender.java [hunk 2]
```diff
@@ -67,26 +67,6 @@ private void init(String slackHookUrl,
                       String slackChannel,
                       Integer queueSize,
                       Integer intervalInMillis) {
-        httpClient = HttpClient.newHttpClient();
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/log/SlackAppender.java:70]: We might have a config object, but that doesn't mean it'll have slack configuration properties. So we're moving these inside the `if` block below.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java [hunk 3]
```diff
@@ -36,7 +36,10 @@
     // https://bugs.mysql.com/bug.php?id=91112
     // Note 03/10/2020: Release of fixed Java connector is imminent and this might no longer be necessary
     static {
-        TimeZone.setDefault(TimeZone.getTimeZone(ConfigManager.getInstance().getConfig().getString(ConfigFile.DEFAULT_TIMEZONE)));
+        ConfigManager manager = ConfigManager.getInstance();
+        if (manager != null && manager.getConfig() != null) {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java:40]: Another `static` block that uses the config manager, so we handle null here.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/security/AesUtil.java [hunk 4]
```diff
@@ -159,10 +159,12 @@ public static void main(String[] args) throws Exception {
 
         if (cmd.hasOption('d')) {
             System.out.println(AesUtil.decrypt(cmd.getOptionValue('s'), encryptionSecret));
+            return;
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/security/AesUtil.java:162]: Each flag in AesUtil does one thing, so we return after finishing that one thing.

**Comment by yufengwng** [comment]
Some screenshots to illustrate (CC @bskinner):

---

Before:
Screen Shot 2021-04-12 at 7 27 49 PM

---

After:
Screen Shot 2021-04-12 at 7 25 18 PM

**Review by ssettipalli — APPROVED** [review]
looks good to me

**Pull request merged**